crates/network/src/transaction/builder.rs

```diff
@@ -196,5 +193,8 @@ pub trait TransactionBuilder<N: Network>: Default + Sized + Send + Sync + 'stati
     fn build_unsigned(self) -> BuilderResult<N::UnsignedTx>;
 
     /// Build a signed transaction.
-    async fn build<S: NetworkSigner<N>>(self, signer: &S) -> BuilderResult<N::TxEnvelope>;
+    fn build<S: NetworkSigner<N>>(
+        self,
+        signer: &S,
+    ) -> impl std::future::Future<Output = BuilderResult<N::TxEnvelope>> + Send;
```

Comment:
does this work for wasm?

Reply:
unsure, but this entire crate does not yet work for wasm because of https://github.com/alloy-rs/alloy/issues/262

Reply:
ah, I assume it's not, `wasm-bindgen` does not generate `Send` futures. i assume we would need to conditionally compile this, but I'd leave that for the above issue perhaps, wdyt

Reply:
related: `impl_future!` macro in `alloy_transport` https://github.com/alloy-rs/alloy/blob/86027c9bb984f3a12a30ffd2a3c5f2f06595f1d6/crates/transport/src/lib.rs#L61